### PR TITLE
Contracts add wasm ink contract in test

### PR DESCRIPTION
**Pull request by pgherveou**
Add an Ink! compiled contract to the test suite

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3173206

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3173776

File: frame/contracts/src/tests.rs [hunk 1]
```diff
@@ -599,6 +604,53 @@ fn migration_on_idle_hooks_works() {
...
+
+	let hex_selector = &spec.constructors[0].selector;
+	let selector = u32::from_str_radix(&hex_selector[2..], 16).expect("Invalid selector");
+	let data = selector.to_be_bytes().encode().to_vec();
+
+	ExtBuilder::default().build().execute_with(|| {
+		let _ = Balances::deposit_creating(&ALICE, 100_000_000_000_000);
+		assert_ok!(Contracts::instantiate_with_code(
+			RuntimeOrigin::signed(ALICE),
+			100_000,
```
Comment (athei) [inline frame/contracts/src/tests.rs:644]: You can't sent balance to an unpayable constructor.

```suggestion
			0,
```

Comment (pgherveou) [reply frame/contracts/src/tests.rs:644]: 🤦

**Comment by pgherveou** [comment]
will close this as @athei  mentioned that we should not have dependencies to ink in the pallet-contracts code (see comment in #14565

**Pull request closed without merging**